doc/overlays.xml

```diff
@@ -17,42 +17,55 @@ if multiple layers override the same package.</para>
 <section xml:id="sec-overlays-install">
 <title>Installing overlays</title>
 
-<para>The list of overlays is determined as follows:
+<para>The list of overlays is determined as follows.</para>
 
-<orderedlist>
+<para>Firstly, if an <varname>overlays</varname> argument to the nixpkgs function itself is given,
+then that is used. This can be passed explicitly when importing nipxkgs, for example 
+<literal>import &lt;nixpkgs> { overlays = [ overlay1 overlay2 ] }</literal>.</para>
 
-  <listitem>
-    <para>First, if an <varname>overlays</varname> argument to the nixpkgs function itself is given,
-    then that is used. This can be passed explicitly when importing nipxkgs, for example 
-    <literal>import &lt;nixpkgs> { overlays = [ overlay1 overlay2 ] }</literal>.</para>
+<para>If the overlays argument is not provided explicitly, we look for overlays in a path. The path
+is determined as follows:
 
-    <para>On a NixOS system the value of the <literal>nixpkgs.overlays</literal> option, if present, 
-    is passed to the system Nixpkgs in this way. Note that this does not affect the overlays for
-    non-NixOS operations (e.g. <literal>nix-env</literal>), which are looked up independently.</para>
-  </listitem>
+<orderedlist>
 
   <listitem>
-    <para>Otherwise, if the Nix path entry <literal>&lt;nixpkgs-overlays></literal> exists and is a 
-    directory, then the result is the set of overlays found in that directory, ordered lexicographically.</para> 
+    <para>First, if the Nix path entry <literal>&lt;nixpkgs-overlays></literal> exists, then it is used.</para>
 
     <para>See the section on <literal>NIX_PATH</literal> in the Nix manual for more details on how to 
     set a value for <literal>&lt;nixpkgs-overlays>.</literal></para>
   </listitem>
 
   <listitem>
-    <para>Otherwise, if <filename>~/.config/nixpkgs/overlays/</filename> exists and is a directory, then
-    the result is the set of overlays found in that directory, ordered lexicographically.</para>
+    <para>If one of <filename>~/.config/nixpkgs/overlays.nix</filename> and
+    <filename>~/.config/nixpkgs/overlays/</filename></para> exists, then it is used. 
+    It is an error if both exist.</para>
   </listitem>
 
 </orderedlist>
 </para>
 
-<para>For the second and third options, overlays are extracted from the given directory as files, 
-directories containing a <filename>default.nix</filename>, or symlinks to one of those.</para>
+<para>If the path is a file, then the file is imported as a Nix expression and used as the list of
+overlays.</para>
+
+<para>If the path is a directory, then we take the set of children of the directory and order it
+lexicographically. Then we attempt to extract an overlay from each by:
```

Comment:
nit: … content of the directory, order it lexicographically, and interpret them as overlays by: